telegram/contrib/django.py

```diff
@@ -0,0 +1,151 @@
+# coding: utf-8
+"""
+This module contains a simple integration with django.
+
+To install on your django project, you need to register all bots on settings, add one line on
+your urls and create a register method it will received a dispatcher to add all handlers of
+your bot.
+
+///////// settings.py
+
+TELEGRAM_BOT = [{
+    'token': '{YOUR_BOT_TOKEN}',
+    'register': 'myapp.dispatcher.register',  # register method that will be called
+    'webhook': 'https://mydomain.com/telegram_webhook/%s'  # optional if you use django.sites
+}]
+
+/////////  urls.py file
+
+urlpatterns = [
+    ...
+    url(r'^telegram_webhook/', include('telegram.contrib.django',)),
+    ...
+]
+
+/// myapp/dispatcher.py
+
+from telegram.ext import CommandHandler
+
+def start(bot, update):
+    bot.sendMessage(update.message.chat_id, text='Hi from telegram bot inside django project!')
+
+def help(bot, update):
+    bot.sendMessage(update.message.chat_id, text='This is a help message')
+
+def error(bot, update):
+    bot.sendMessage(update.message.chat_id, text='Command not founded.')
+
+# this method will be called on start of application
+def register(dispatcher):
+    dispatcher.add_handler(CommandHandler("start", start))
+    dispatcher.add_handler(CommandHandler("help", help))
+    dp.add_error_handler(error)
+
+"""
+
+import json
+import logging
+
+try:
+    import django  # noqa
+except ImportError as e:
+    raise Exception('Need Django installed.')
+
+from telegram import Bot
+from telegram import Update
+from telegram.ext import Dispatcher
+
+from django.conf.urls import url
+from django.http import HttpResponse
+from django.views import generic
+from django.conf import settings
+from django.http.response import Http404
+from django.utils.module_loading import import_string
+from django.core.urlresolvers import reverse
+from django.utils.decorators import method_decorator
+from django.views.decorators.csrf import csrf_exempt
+
+logger = logging.getLogger(__name__)
+
+BOTS_REGISTERED = {}
+
+
+@method_decorator(csrf_exempt, name='dispatch')
+class TelegramView(generic.View):
+
+    @classmethod
+    def as_view(cls, **initkwargs):
+        register_webhooks()
+        return super(TelegramView, cls).as_view(**initkwargs)
+
+    def get(self, request):
+        return HttpResponse()
+
+    def head(self, request):
+        return HttpResponse()
+
+    def post(self, request, token):
+        dispatcher = self.get_dispatcher(token)
+        if not dispatcher:
+            return Http404()
+
+        json_string = request.body.decode('utf-8')
+        update = Update.de_json(json.loads(json_string))
+        dispatcher.process_update(update)
+        return HttpResponse()
+
+    def get_dispatcher(self, token):
+        dispatcher = None
+        if token in BOTS_REGISTERED:
+            return BOTS_REGISTERED[token]
+        return dispatcher
+
+
+class DjangoDispatcher(Dispatcher):
```

Comment:
There is no point of creating custom dispatcher, just use the default one with update_queue=None, workers=0